core/src/avm2/multiname.rs

```diff
@@ -22,28 +57,51 @@ use swf::avm2::types::{
 #[collect(no_drop)]
 pub struct Multiname<'gc> {
     /// The list of namespaces that satisfy this multiname.
-    ns: Vec<Namespace<'gc>>,
+    ns: MultinameNamespaceSet<'gc>,
 
     /// The local name that satisfies this multiname. If `None`, then this
     /// multiname is satisfied by any name in the namespace.
     name: Option<AvmString<'gc>>,
 
     /// The type parameters required to satisfy this multiname. If empty, then
     /// this multiname is satisfied by any type parameters in any amount.
-    params: Vec<Multiname<'gc>>,
+    params: Vec<Gc<'gc, Multiname<'gc>>>,
+
+    /// Whether the namespace needs to be read at runtime before use.
+    /// This should only be `false` when lazy-initialized in Activation.
+    has_lazy_ns: bool,
+    /// Whether the name needs to be read at runtime before use
+    /// This should only be `false` when lazy-initialized in Activation.
+    has_lazy_name: bool,
```

Comment:
Could use `bitflags` to save a byte here, which might add up over a lot of names and make the `multiname.has_lazy_name() && !multiname.has_lazy_ns()` nicer (checking for a specific bit pattern).

Reply:
Only reason I didn't do this was because as-is this extra bool wouldn't affect structure size anyway, due to alignment; but I can pack them into a bit field, sure.